### PR TITLE
refactor(bazel): Leverage new parser in update function

### PR DESCRIPTION
**Pull request by zharinov**
## Context

- Ref: #9667

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/bazel/update.spec.ts [hunk 1]
```diff
@@ -119,25 +119,20 @@ describe('modules/manager/bazel/update', () => {
...
-        `"${currentDigest}"`,
-        `"${newDigest}",  # ${newValue}`
-      );
+      const output = input.replace(`"${currentDigest}"`, `"${newDigest}"`);
 
       const res = await updateDependency({ fileContent: input, upgrade });
 
       expect(res).toEqual(output);
-      expect(res).toContain(
-        '"aaa09d789f3dba190787f8b4454c7d3c936fe123",  # v1.0.3'
```
Comment (zharinov) [inline lib/modules/manager/bazel/update.spec.ts:139]: Do we really need this comment with tag? Seems like it's not worth the code complexity and problems this functionality may cause

File: lib/modules/manager/bazel/common.spec.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { updateCode } from './common';
+
+describe('modules/manager/bazel/common', () => {
+  describe('updateFragment', () => {
```
Comment (viceice) [inline lib/modules/manager/bazel/common.spec.ts:4]:
```suggestion
  describe('updateCode', () => {
```

File: lib/modules/manager/bazel/update.spec.ts (hunk 1 above)
Comment (viceice) [reply lib/modules/manager/bazel/update.spec.ts:139]: i think this is the same behavior which we use at GitHub actions to pin the git commit to a git tag.

so i think yes we need this.

File: lib/modules/manager/bazel/update.ts [hunk 3]
```diff
@@ -1,54 +1,68 @@
-// TODO: types (#7154)
-/* eslint-disable @typescript-eslint/restrict-template-expressions */
+import is from '@sindresorhus/is';
 import hasha from 'hasha';
 import { logger } from '../../../logger';
 import * as packageCache from '../../../util/cache/package';
 import { Http } from '../../../util/http';
+import * as p from '../../../util/promises';
```
Comment (viceice) [inline lib/modules/manager/bazel/update.ts:6]: can we use direct import instead?

File: lib/modules/manager/bazel/update.spec.ts (hunk 1 above)
Comment (zharinov) [reply lib/modules/manager/bazel/update.spec.ts:139]: Here is the first time it was introduced: https://github.com/renovatebot/renovate/pull/3267

Comment (viceice) [reply lib/modules/manager/bazel/update.spec.ts:139]: so yes we need this

Comment (zharinov) [reply lib/modules/manager/bazel/update.spec.ts:139]: I can confirm we've broken the solution to the original problem, so maintaining this comment with tag version makes no sense

Comment (zharinov) [reply lib/modules/manager/bazel/update.spec.ts:139]: Let's focus on this problem later, I think I have better idea than tag commenting...

**Review by rarkins — DISMISSED** [review]

File: lib/modules/manager/bazel/update.spec.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import crypto from 'crypto';
 import * as httpMock from '../../../../test/http-mock';
 import type { UpdateType } from '../../../config/types';
-import { updateDependency } from '.';
+import { updateDependency } from './update';
```
Comment (viceice) [inline lib/modules/manager/bazel/update.spec.ts:4]:
```suggestion
import { updateDependency } from '.';
```
always import from index (branch coverage), or isn't it no longer public

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Review by rarkins — APPROVED** [review]

**Pull request merged**